src/l500/l500-options.cpp

```diff
@@ -63,28 +86,116 @@ namespace librealsense
         auto max_value = float(*(reinterpret_cast<int32_t*>(max.data())));
         auto min_value = float(*(reinterpret_cast<int32_t*>(min.data())));
 
+        bool success;
+        auto res = query_default( success );
+
+        if( !success )
+        {
+            _is_read_only = true;
+            res = -1;
+        }
+
         _range = option_range{ min_value,
-            max_value,
-            float(*(reinterpret_cast<int32_t*>(step.data()))),
-            def };
+                               max_value,
+                               float( *( reinterpret_cast< int32_t * >( step.data() ) ) ),
+                               res };
+    }
+
+    void l500_hw_options::set_default( float def )
+    {
+        _range.def = def;
+    }
+
+    void l500_hw_options::set_read_only( bool read_only )
+    {
+        _is_read_only = read_only;
+    }
+
+    void l500_hw_options::set_manually( bool set ) 
+    {
+        _was_set_manually = set;
     }
 
     void l500_hw_options::enable_recording(std::function<void(const option&)> recording_action)
     {}
 
-    float l500_hw_options::query(int mode) const
+    // this method can throw an exception if un-expected error occurred
+    // or return error by output parameter 'success' if default value not applicable
+    // on this state return value is undefined
+    float l500_hw_options::query_new_fw_default( bool & success ) const
     {
-        auto res = _hw_monitor->send(command{ AMCGET, _type, get_current, mode });
-
-        if (res.size() < sizeof(int32_t))
+        success = true;
+        hwmon_response response;
+        auto res = _hw_monitor->send( command{ AMCGET,
+                                               _type,
+                                               l500_command::get_default,
+                                               (int)query_sensor_mode( *_resolution ) },
+                                      &response );
+
+        // If of digital gain is on 'auto' the gain value can change in FW internally
```

Comment:
// Controls that are automatically set by the FW (e.g., APD when digital gain is AUTO) are read-only,
// cannot be modified, and have no defaults: the FW will return hwm_IllegalHwState for these...